src/BayesFilters/include/BayesFilters/utils.h

```diff
@@ -83,6 +83,102 @@ Eigen::VectorXd multivariate_gaussian_log_density(const Eigen::MatrixBase<Derive
 }
 
 
+/**
+ * Evaluate the logarithm of a multivariate Gaussian probability density function
+ * with high-dimensional input using the Sherman–Morrison–Woodbury formula.
+ *
+ * It is assumed that the covariance matrix S can be written in the form S = UV + R
+ * where R is an invertible block diagonal matrix. Each block R_{i} is square and has
+ * dimension M such that N * M is the size of the input for some positive integer N.
+ *
+ * This version is much faster than the standard gaussian logarithm density evaluation
+ * if U.cols() << U.rows().
+ *
+ * @param input Input representing the argument of the function as a vector or matrix.
+ * @param mean The mean of the associated Gaussian distribution as a vector.
+ * @param U The U factor within the covariance matrix S = UV + R of the associated Gaussian distribution as a matrix.
+ * @param V The V factor within the covariance matrix S = UV + R of the associated Gaussian distribution as a matrix.
+ * @param R The R summand within the covariance matrix S = UV + R of the associated Gaussian distribuion as a matrix.
+ *        The matrix R has size M * (N * M) and consists in the concatenation of all the diagonal blocks R_{i}.
```

Comment:
Please indent this and the following line with the previous one starting from `The R summand ...` 😄

Reply:
Ok :+1: